astropy/utils/decorators.py

```diff
@@ -834,12 +834,14 @@ def __get__(self, obj, owner=None):
                     if val is _NotFound:
                         val = self.fget(obj)
                         obj_dict[self._key] = val
-            return val
+            
```

Comment:
same comment here: if we're refactoring the `try` blocks then we should go all in, as the Python docs https://docs.python.org/3/tutorial/errors.html indicate

> "The use of the else clause is better than adding additional code to the [try](https://docs.python.org/3/reference/compound_stmts.html#try) clause because it avoids accidentally catching an exception that wasn’t raised by the code being protected by the try … except statement."

And only have the bit of code that might raise the AttributeError in the `try` statement, all else in the `else`.